src/mqtt_client/AtlasMqttClient.cpp

```diff
@@ -0,0 +1,78 @@
+#include "AtlasMqttClient.h"
+#include "AtlasMqttClient_callback.h"
+#include "AtlasMqttClient_actionListener.h"
+#include "AtlasMqttLogger.h"
+#include "AtlasMqttException.h"
+#include <mqtt/async_client.h>
+
+namespace Atlas
+{
+AtlasMqttClient* AtlasMqttClient::instance = nullptr;
+
+AtlasMqttClient& AtlasMqttClient::getInstance()
+{
+    if (instance == nullptr)
+    {
+        instance = new AtlasMqttClient;
+    }
+    return *instance;
+}
+
+void AtlasMqttClient::destroyInstance()
+{
+    if (instance != nullptr)
+    {
+        delete instance;
+        instance = nullptr;
+    }
+}
+
+void AtlasMqttClient::publishMessage(const string address, const string clientID, const string topic, 
+                    const string message, int QoS)
+{
+    AtlasMqttLogger::getInstance().logMessage("Initializing client (" + clientID + 
+                        ") connection to server (" + address + ")");
+
+    mqtt::async_client client(address, clientID);
```

Comment:
This will establish a new connection on every call. Can we cache this connection? For subscribe we need an open socket anyway. If yes, what happens if the network connection breaks - will the async client re-establish the connection?

Reply:
Made the connection a shared object as a class member. For now, there isn't a reconnecting logic implemented in the code.